File: src/main/java/org/gangel/graphomance/arangodb/ArangoSession.java
package org.gangel.graphomance.arangodb;

import com.arangodb.ArangoDatabase;
import lombok.Builder;
import lombok.Getter;
import org.gangel.graphomance.ManagementApi;
import org.gangel.graphomance.ObjectApi;
import org.gangel.graphomance.SchemaApi;
import org.gangel.graphomance.Session;
import org.gangel.graphomance.metrics.ObjectApiMetricsWrapper;

@Builder
public class ArangoSession implements Session {

    @Getter
    private ArangoDatabase db;

    @Getter
    private ArangoConnectionSettings connectionSettings;

    @Override
    public void close() {
    }

    @Override
    public SchemaApi schemaApi() {
        return ArangoSchemaApi.builder().db(db).build();
    }

    @Override
    public ObjectApi objectApi() {
        return ObjectApiMetricsWrapper.create(ArangoObjectApi.builder().db(db).build());
    }

    @Override
    public ManagementApi managementApi() {
        return ArangoManagementApi.builder().db(db).build();
    }

}

File: src/main/kotlin/org/gangel/graphomance/arangodb/ArangoSession.kt
package org.gangel.graphomance.arangodb

import com.arangodb.ArangoDatabase
import org.gangel.graphomance.ObjectApi
import org.gangel.graphomance.SchemaApi
import org.gangel.graphomance.Session
import org.gangel.graphomance.metrics.ObjectApiMetricsWrapper

class ArangoSession(
	val db: ArangoDatabase,
	val connectionSettings: ArangoConnectionSettings
) : Session {

	override fun close() {}
	override fun schemaApi(): SchemaApi {
		return ArangoSchemaApi(db = db)
	}

	override fun objectApi(): ObjectApi {
		return ObjectApiMetricsWrapper.create(ArangoObjectApi(db = db))
	}

}